### ISSUE TITLE
[Crash] Subs with custom preview images crash when attempting to upload to workshop

### ISSUE THREAD

**Issue by hubismal** [Regalis11/Barotrauma#1400]
I'm trying to publish my absolutely hilarious sub "The Maturity" to the workshop, using this image:
![MaturityLightingScrshot](https://user-images.githubusercontent.com/47284081/55688464-6a6ec700-593e-11e9-9caf-9d90648ad70b.png)
Unfortunately, the game seems to crash and immediately close out when I try and upload (aka the moment I hit share and it says the "uploading to..." dialogue for a second). If I had to take a guess it's due to an issue with file type maybe? This is a pretty high res PNG here.
Here's the crashlog:
[crashreport.log](https://github.com/Regalis11/Barotrauma/files/3051953/crashreport.log)

**Comment by hubismal** [comment]
Additional info: The game sets the preview just fine and it works great ingame.
![2019-04-07-135940_1366x768_scrot](https://user-images.githubusercontent.com/47284081/55688495-be79ab80-593e-11e9-802d-257aee491bdb.png)
It's just the uploading part that doesn't work.

**Comment by Regalis11** [comment]
>Exception: PreviewImage should be under 1MB (1161851)

The maximum file size for the preview images is 1 MB. The game checks the file size when you're setting the image through the workshop publish menu, but looks like the check is skipped when using the submarine's preview image. Got to add it there as well, thanks for pointing this out!

**Comment by hubismal** [comment]
Even more additional info:
Rokvach says the error is due to the filesize exceeding 1MB.
The original image, MaturityLightingScreenshot.png, made in GIMP, is 747.2KB. But when it's uploaded to the game, the game creates a slightly different image (you can see a difference with the transparency resolution if you view them side by side) named PreviewImage.png that sizes up to 1.1MB for some reason.

**Comment by hubismal** [comment]
tl;dr, the game increases the filesize of the image for reasons unknown.

**Comment by Regalis11** [comment]
You could probably work around that by changing the image when publishing it to workshop (i.e. instead of using the preview image loaded from the sub file, click "Browse" and find the original file).

**Comment by Regalis11** [comment]
Fixed in 2f2a6df

**Comment by Regalis11** [comment]
Fixed in 532e8bf
